packages/server/src/core/internals/utils.ts

```diff
@@ -4,13 +4,17 @@ import { ProcedureParams } from '../procedure';
 /**
  * @internal
  */
-export type Overwrite<TType, TWith> = {
-  [K in keyof TType | keyof TWith]: K extends keyof TWith
-    ? TWith[K]
-    : K extends keyof TType
-    ? TType[K]
-    : never;
-};
+export type Overwrite<TType, TWith> = TType extends any
+  ? TWith extends any
+    ? {
+        [K in keyof TType | keyof TWith]: K extends keyof TWith
+          ? TWith[K]
+          : K extends keyof TType
+          ? TType[K]
+          : never;
+      }
+    : never
+  : never;
```

Comment:
This shouldn't really have any cost, but causes typescript to correctly distribute the union elements again as per here: https://stackoverflow.com/a/51691257